src/python/pants/option/options_bootstrapper.py

```diff
@@ -213,7 +214,7 @@ def get_bootstrap_options(self) -> Options:
         return self.bootstrap_options
 
     @memoized_method
-    def _full_options(self, known_scope_infos: Tuple[ScopeInfo, ...]) -> Options:
+    def _full_options(self, known_scope_infos: FrozenOrderedSet[ScopeInfo]) -> Options:
```

Comment:
This cannot be `Iterable` or `Sequence` because `@memoized_method` requires a hashable type so passing something like a list—then relying on the downstream implementation to sort—would fail.